remove-webhook/adjust-downstream-webhook.sh

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env bash
+
+usage() {
+  cat << EOF
+usage: bash adjust-downstream-webhook.sh [--insecure-skip-tls-verify]
+
+This script adjusts the rancher-webhook chart release in all clusters managed by Rancher (excluding the local cluster).
+Depending on the version of Rancher, it either deletes the downstream webhook release, adjusts its version and restarts, or does nothing.
+Requires kubectl and helm to be installed and available on \$PATH.
+
+RANCHER_URL without a trailing slash must be set with the server URL of Rancher.
+RANCHER_TOKEN must be set with an admin token generated with no scope.
+To ignore TLS verification, set --insecure-skip-tls-verify.
+
+Users also need to ensure they have the rancher-charts repo in the local Helm index.
+EOF
+}
+
+if [ "$1" == "-h" ]; then
+  usage
+  exit 0
+fi
+
+delete_webhook() {
+  cluster="$1"
+  current_chart=$(helm list -n cattle-system -l name=rancher-webhook | tail -1 | cut -f 6)
+  echo "Deleting $current_chart from cluster $cluster."
+  helm uninstall rancher-webhook -n cattle-system
+}
+
+replace_webhook() {
+  cluster="$1"
+  new_version="$2"
+
+  echo "Updating the agent to make it remember the min version $new_version of rancher-webhook, so that it can deploy it when needed in the future in cluster $cluster."
+  kubectl set env -n cattle-system deployment/cattle-cluster-agent CATTLE_RANCHER_WEBHOOK_MIN_VERSION="$new_version"
+
+  helm get values -n cattle-system rancher-webhook -o yaml > current_values.yaml
+  echo "Re-installing rancher-webhook to use $new_version in cluster $cluster."
+  helm upgrade --install rancher-webhook rancher-charts/rancher-webhook -n cattle-system --version "$new_version" --values current_values.yaml
```

Comment:
I think cattle-cluster-agent will restart after line 36 and it will reinstall the webhook itself with the new version? Or is that not how it works?

Reply:
Correct, the agent will restart but won't decide to re-install the webhook because of the min version being lower or equal to the one already installed. This is the logic of our system chart manager.